packages/web3-eth/test/integration/defaults.test.ts

```diff
@@ -0,0 +1,763 @@
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import WebSocketProvider from 'web3-providers-ws';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Contract } from 'web3-eth-contract';
+import { hexToNumber, numberToHex } from 'web3-utils';
+import { TransactionBuilder, TransactionTypeParser, Web3Context } from 'web3-core';
+import { DEFAULT_RETURN_FORMAT } from 'web3-common';
+import { prepareTransactionForSigning, transactionBuilder, Web3Eth } from '../../src';
+
+import { createNewAccount, getSystemTestProvider } from '../fixtures/system_test_utils';
+import {
+	defaultTransactionBuilder,
+	getTransactionFromAttr,
+	getTransactionType,
+} from '../../src/utils';
+import { BasicAbi, BasicBytecode } from '../shared_fixtures/build/Basic';
+import { ERC20TokenAbi, ERC20TokenBytecode } from '../shared_fixtures/build/ERC20Token';
+import { detectTransactionType } from '../../dist';
+import { getTransactionGasPricing } from '../../src/utils/get_transaction_gas_pricing';
+import { sendFewTxes } from './helper';
+
+describe('defaults', () => {
+	let web3Eth: Web3Eth;
+	let accounts: string[] = [];
+	let clientUrl: string;
+	let contract: Contract<typeof BasicAbi>;
+	let contractERC20: Contract<typeof BasicAbi>;
+	let deployOptions: Record<string, unknown>;
+	let sendOptions: Record<string, unknown>;
+
+	beforeAll(async () => {
+		clientUrl = getSystemTestProvider();
+		const acc1 = await createNewAccount({ unlock: true, refill: true });
+		const acc2 = await createNewAccount({ unlock: true, refill: true });
+		accounts = [acc1.address, acc2.address];
+		web3Eth = new Web3Eth(clientUrl);
+
+		contract = new Contract(BasicAbi, undefined, undefined, web3Eth.getContextObject() as any);
+		deployOptions = {
+			data: BasicBytecode,
+			arguments: [10, 'string init value'],
+		};
+		sendOptions = { from: accounts[0], gas: '1000000' };
+		contract = await contract.deploy(deployOptions).send(sendOptions);
+
+		contractERC20 = new Contract(
+			ERC20TokenAbi,
+			undefined,
+			undefined,
+			web3Eth.getContextObject() as any,
+		);
+		contractERC20 = await contractERC20
+			.deploy({
+				data: ERC20TokenBytecode,
+				arguments: ['1000000000000000000'],
+			})
+			.send({ from: accounts[1], gas: '2700000' });
+	});
+	afterAll(() => {
+		if (clientUrl.startsWith('ws')) {
+			(web3Eth.provider as WebSocketProvider).disconnect();
+		}
+	});
+
+	describe('defaults', () => {
+		it('defaultAccount', async () => {
+			// default
+			expect(web3Eth.defaultAccount).toBeUndefined();
+
+			// after set
+			web3Eth.setConfig({
+				defaultAccount: accounts[0],
+			});
+			expect(web3Eth.defaultAccount).toBe(accounts[0]);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
```

Comment:
Doesn't this handle on ws? Shouldn't it be disconnected as `eth` inside `afterAll`?
As I see, in most tests eth2 is present, maybe it should be better to disconnect inside `afterAll`.

Reply:
thx. fixed